terminus/generators/monolane_generator.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import print_function
+
+from collections import OrderedDict
+from math import atan2
+from math import degrees
+import yaml
+
+from .city_visitor import CityVisitor
+from models.intersection_node import IntersectionNode
+
+
+class MonolaneGenerator(CityVisitor):
+    def start_city(self, city):
+        """Called when a visitation of a new city starts."""
+        # Reset all of the class variables used until stop_city is called.
+        self.monolane = {
+            'maliput_monolane_builder': OrderedDict([
+                ('id', city.name),
+                ('lane_bounds', [-2, 2]),
+                ('driveable_bounds', [-4, 4]),
+                ('position_precision', 0.01),
+                ('orientation_precision', 0.5),
+                ('points', OrderedDict()),
+                ('connections', OrderedDict()),
+                ('groups', OrderedDict()),
+            ])
+        }
+        self.streets = []
+        self.point_name_by_street_node_dict = {}
+        self.intersection_index = 0
+        self.intersection_index_by_node = {}
+
+    def node_ident(self, street, node):
```

Comment:
The usage of a visitor for that case came from RNDF, as it maps ids in a hierarchical fashion (`segment_id.lane_id.waypoint_id`) so using the visitor was a natural way of mapping that recursive path. Agree 100% though that, if that is not a requirement, there is no need to implement it as a visitor (and I'm not 100% positive that it can be generalized either :)). I just saw that a similar requirement (doing a first pass to generate an id->object lookup table) was present in two generators before, so I thought it would be a good think to separate them and see if we can find at least (as you mention) a common interface. With that first step we can see if we can later reuse some behavior.